gradunwarp/core/unwarp_resample.py

```diff
@@ -199,167 +202,175 @@ def non_linear_unwarp_siemens(self, volshape, dv, dxyz, m_rcs2lai, m_rcs2lai_noh
         vc, vr = utils.meshgrid(np.arange(nc), np.arange(nr))
 
         # Compute transform to map the internal voxel coordinates to FSL scaled mm coordinates
-        try:
-            pixdim1=float((subprocess.Popen(['fslval', self.name,'pixdim1'], stdout=subprocess.PIPE).communicate()[0]).strip())
-        except ValueError:
-            log.error('Failure during fslval call. Make sure fslval, fslhd, fslorient are in your PATH, and that FSLOUTPUTTYPE is set.')
-            sys.exit(1)
-        
-        pixdim2=float((subprocess.Popen(['fslval', self.name,'pixdim2'], stdout=subprocess.PIPE).communicate()[0]).strip())
-        pixdim3=float((subprocess.Popen(['fslval', self.name,'pixdim3'], stdout=subprocess.PIPE).communicate()[0]).strip())
-        dim1=float((subprocess.Popen(['fslval', self.name,'dim1'], stdout=subprocess.PIPE).communicate()[0]).strip())
-        outputOrient=subprocess.Popen(['fslorient', self.name], stdout=subprocess.PIPE).communicate()[0].strip()
-        if outputOrient == b'NEUROLOGICAL':
-            log.info('Input volume is NEUROLOGICAL orientation. Flipping x-axis in output fullWarp_abs.nii.gz')
+        pixdim1, pixdim2, pixdim3 = self.input_nii.header.get('pixdim')[1:4]
+
+        dim1 = self.input_nii.header.get('dim')[1]
+        axcodes = ''.join(nib.orientations.aff2axcodes(self.input_nii.affine))
+
+        # TODO: figure out if that logic works
+        outputOrient = 'NEUROLOGICAL' if axcodes in ['RAS', 'RPI', 'LPS', 'LAI'] else 'RADIOLOGICAL'
```

Comment:
This should probably check the determinant of the 3x3 part of the sform rather than checking against a list (half of the 48 plumb possibilities are left-handed, though other parts of the code may misbehave on many of them for other reasons).  fslorient appears to check the 3x3 determinant of both sform and qform, and additionally has an "inconsistent" code when they don't match.

Reply:
Thanks for looking into this. Could you provide a link to the fsl repo of where you found the logic used in fslorient you described?

Reply:
I ended up finding it here via google, though it seems to be an old version:

https://git.mpib-berlin.mpg.de/krause/fsl/-/blob/8b58b9a0db0a7b07f725fb97d56ea6f6d7b89c43/src/fslio/fslio.c#L1774

I tried to look through the fmrib gitlab for it, but the fsl tools and libraries are now extremely fragmented across repositories (7 pages of repo names?), and its search doesn't return any code results for anything, so I gave up...

Reply:
Thanks, I guess that the logic has not evolved in more recent version.
I implemented it in a separate function.